openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIRequestBodyMapper.java

```diff
@@ -123,8 +121,16 @@ public void handlePayloadAnnotation(RequiredParameterNode payloadNode, Map<Strin
             if (mapMime != null) {
                 fields = mapMime.fields();
             }
+
+            // Handle multiple mime type when attached with request payload annotation.
+            // ex: <pre> @http:Payload { mediaType:["application/json", "application/xml"] } Pet payload </pre>
             if (fields != null && (!fields.isEmpty() || fields.size() != 0)) {
```

Comment:
Why do we need to keep both `!fields.isEmpty()` and `fields.size() != 0` checks in here?

Reply:
Yeah, we can remove one condition there. we earlier added this due to an issue with API. as an example we have `[ , ]` array this is not empty but there is no item. since these issues have already been fixed from API. I will remove unnecessary checks here. 🙂